api/api.go

```diff
@@ -147,6 +170,10 @@ func (b *BlockChainAPI) SendRawTransaction(
 
 // GasPrice returns a suggestion for a gas price for legacy transactions.
 func (b *BlockChainAPI) GasPrice(ctx context.Context) (*hexutil.Big, error) {
+	if err := rateLimit(ctx, b.limiter, b.logger); err != nil {
+		return nil, err
+	}
+
 	return (*hexutil.Big)(b.config.GasPrice), nil
```

Comment:
Same here, that's a fairly cheap and simple endpoint. No DB interactions.

Reply:
I can remove on the simple ones.